### ISSUE TITLE
Insert regression with commit 8051664, can not restore anymore a graph from protobuf

### ISSUE THREAD

**Issue by lucaRadicalbit** [eaplatanios/tensorflow_scala#97]
In the commit [8051664](https://github.com/eaplatanios/tensorflow_scala/commit/805166420a46be534244e9fe43cd4a1777c8f07b#diff-27b966693fad665d3134d035ccb9c928) I see that in the `fromProto` method the `graphElement` is taken adding the `handle.op.name` to the `/Read/ReadVariable:0` String. But restoring the graph the name in the protobuf is:
```
node {
    name: "foo_weights_w/Read/ReadVariableOp"     < - - - - - - - - - - - - - - - -
    op: "ReadVariableOp"
    input: "foo_weights_w"
    device: "/device:CPU:0"
    attr {
      key: "dtype"
      value {
        type: DT_DOUBLE
      }
    }
    attr {
      key: "_class"
      value {
        list {
          s: "loc:@foo_weights_w"
        }
      }
    }
    attr {
      key: "_output_shapes"
      value {
        list {
          shape {
            dim {
              size: 1
            }
            dim {
              size: 1
            }
          }
        }
      }
    }
  }
```

So I wondering if the `/Read/ReadVariable:0` String shuldn't be `/Read/ReadVariableOp:0` String

**Comment by eaplatanios** [comment]
@lucaRadicalbit Does that refer to a graph saved with a previous version of the API or one both saved and restored with the current master?

**Comment by lucaRadicalbit** [comment]
@eaplatanios No, I saved the graph with the Python API and then restored the graph the `0.2.0-SNAPSHOT`version. Il the first time I have this problem, with the other version was everything ok.

**Comment by eaplatanios** [comment]
@lucaRadicalbit Ok. Let me look into this. It's just a matter of changing the default name of the read op.

**Comment by eaplatanios** [comment]
@lucaRadicalbit Could you please confirm that this has now been fixed? Thanks!

**Comment by lucaRadicalbit** [comment]
@eaplatanios I can not investigate further today but just to get you an answer the previous error is gone but now when I run my app I get:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00000001251f0471, pid=2295, tid=0x0000000000013e03
#
# JRE version: Java(TM) SE Runtime Environment (8.0_151-b12) (build 1.8.0_151-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.151-b12 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# [thread 50435 also had an error]
C  [libtensorflow_framework.so+0x52b471]  nsync::nsync_mu_lock(nsync::nsync_mu_s_*)+0x11
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# ..../hs_err_pid2295.log
[thread 81155 also had an error]
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

In the next days I'll try to take a deeply look.

**Comment by lucaRadicalbit** [comment]
At the moment I don't know why but in my project I still have your code without the bug fix done in the commit a084e85. I also try to clean the cache and reload everything.
